### ISSUE TITLE
Online waiver

### ISSUE THREAD

**Issue by willgearty** [learning-unlimited/ESP-Website#2572]
New module that would allow for parents/students to sign a liability waiver through the website (or 3rd party website via API).

Requested by Stanford

**Comment by benjaminjkraft** [comment]
This could be similar to the existing Formstack module, although we might need to use a different vendor that does fancy esign stuff.

**Comment by willgearty** [comment]
Yeah, that's what I was thinking. As someone in in the tech sector, any suggestions for specific vendors @benjaminjkraft ?

**Comment by benjaminjkraft** [comment]
Not specifically.  I think this is the kind of thing where the school is likely to dictate a vendor.  If they don't, or if there's a choice, go for one that mentions their API as prominently as possible, since that is a good indicator that they consider it an important feature and put effort into making it usable.

**Comment by HiddendarkViper07** [comment]
@willgearty I'm proposing to implement the Online Waiver module with DocuSign integration.

### Proposed Implementation:
- **Vendor**: DocuSign (free for educational institutions, robust API)
- **Models**: `WaiverTemplate` (stores waiver content) and `WaiverSignature` (tracks signed records with audit trail)
- **Flow**: User signs via DocuSign → webhook updates status in system → registration flow continues
- **API**: Endpoints to check signing status and download signed PDFs
- **Admin**: Interface to view, filter, and download all signed waivers

Would appreciate any feedback or suggestions . Thanks!

**Comment by willgearty** [comment]
I can't find any documentation for a free educational institution plan for DocuSign. @HiddendarkViper07

**Comment by HiddendarkViper07** [comment]
@willgearty ,Thanks for pointing that out — I couldn’t find confirmation of a free DocuSign education plan either.

Instead, I’ll design this as a vendor-agnostic module so schools can use any e-sign provider (DocuSign, Dropbox Sign, etc.), with a simple internal fallback if none is configured.

Would that approach work better?

Also, would it be helpful if I build a small demo module/prototype and include it in my proposal for better clarity?

**Comment by willgearty** [comment]
We're not ready to move forward with this right now (hence the Idea for Discussion tag).

**Comment by HiddendarkViper07** [comment]
Thanks for the clarification!

No problem — I understand it's not a priority right now. I’ll focus on other areas and keep this idea in mind for the future.

Thanks!
